bin/bib_rights.pl

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use utf8;
+
+BEGIN {
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/cgi');
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/post_zephir_processing');
+}
+
+use Encode;
+use Getopt::Long qw(:config no_ignore_case bundling);
+use Term::ANSIColor qw(:constants colored);
+$Term::ANSIColor::AUTORESET = 1;
+use Data::Dumper;
+use File::Slurp;
+use MARC::Record;
+use MARC::File::XML(BinaryEncoding => 'utf8');
+
+use bib_rights;
+use CRMS;
+
+binmode(STDOUT, ':encoding(UTF-8)');
+
+my $usage = <<END;
+USAGE: $0 [-hp] [-i path/to/file] [htid_or_cid_1 [htid_or_cid_2 ...]]
+
+Reports the bibliographic rights for one or more HathiTrust and/or catalog ids.
+
+-h, -?    Print this help message.
+-i        Input file with whitespace-delimited (space, tab, newline) list of ids.
+-v        Report all fields in the bib_rights_info data structure.
+-y YEAR   Use YEAR instead of the current one (for predicting new year rollover).
+END
+
+
+my $help;
+my $infile;
+my $verbose;
+my $year;
+
+Getopt::Long::Configure('bundling');
+die 'Terminating' unless GetOptions(
+  'h|?'  => \$help,
+  'i:s'  => \$infile,
+  'v+'   => \$verbose,
+  'y:s'  => \$year);
+
+if ($help) { print $usage. "\n"; exit(0); }
+
+my $crms = CRMS->new(
+  instance => 'production'
+);
+
+
+if (defined $year) {
+  $ENV{BIB_RIGHTS_DATE} = $year;
+}
+
+my @ids;
+
+if (defined $infile) {
+  unless (-e $infile) {
+    die "Error: can't read $infile";
+  }
+  my $text = File::Slurp::read_file($infile);
+  push @ids, $_ for split(/\s+/, $text);
+}
+
+push @ids, $_ for @ARGV;
+
+my $bib_rights = BibRights->new;
+foreach my $id (@ids) {
+  my $res = $bib_rights->query($id);
+  if ($res->{error}) {
+    print BOLD "$id $res->{error}\n";
+    next;
+  }
+  # Show catalog id as separate line if it was a cid query.
+  if ($id eq $res->{cid}) {
+    print BOLD BLUE "$id\n";
+  }
+  DisplayEntry($id, $_) for @{ $res->{entries} };
+  if (scalar @ids > 1) {
+    printf "%s\n", '_' x 20;
```

Comment:
likewise these separators make it slightly more annoying to process the output from this